app/Http/Requests/AdminNewUpdateMarketRequest.php

```diff
@@ -24,9 +25,21 @@ public function authorize()
      */
     public function rules()
     {
+        // Get the ID off the route
+        $id = $this->route('id') ?? null;
+        $sponsor_id = $this->only('sponsor');
```

Comment:
does the only method plucks the id of whatever var name we're passing in there? nice

Reply:
if it's the Laravel collections method only - https://laravel.com/docs/6.x/collections#method-only - i think your argument here should be `sponsor_id` if you need the id of the sponsor. same for market in the AdminNewUpdateTraderRequest

Reply:
`$this` refers to the formRequest, which is a type of Request. only, will pull the input off. 
However, you may be right, i don't think i want an array, which only returns.